### PR TITLE
Add source in “error from server” message when using kubectl

### PR DESCRIPTION
**Pull request by feihujiang**
#10212

### CODE REVIEW

File: pkg/kubectl/cmd/create.go [hunk 1]
```diff
@@ -97,11 +97,11 @@ func RunCreate(f *cmdutil.Factory, out io.Writer, filenames util.StringList) err
 	err = r.Visit(func(info *resource.Info) error {
 		data, err := info.Mapping.Codec.Encode(info.Object)
 		if err != nil {
-			return err
+			return fmt.Errorf("error when creating %q: %v ", info.Source, err)
```
Comment (nikhiljindal) [inline pkg/kubectl/cmd/create.go]: If I am not wrong, this should be an api/error (https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/errors/errors.go#L91).

cc @bgrant0607 @lavalamp

**Comment by jlowdermilk** [comment]
@k8s-bot ok to test

File: pkg/kubectl/cmd/create.go (hunk 1 above)
Comment (jlowdermilk) [inline pkg/kubectl/cmd/create.go]: Shouldn't be necessary. The error returned here gets handled in [`CheckErr`](https://github.com/GoogleCloudPlatform/kubernetes/pull/10221/files#diff-a0580db929d5a6aeb59e7473a86017adR54).

**Comment by jlowdermilk** [comment]
LGTM. Seems like strict improvement to usability.

File: pkg/kubectl/resource/visitor.go [hunk 2]
```diff
@@ -65,6 +65,7 @@ type Info struct {
 	Mapping   *meta.RESTMapping
 	Namespace string
 	Name      string
+	Source    string
```
Comment (lavalamp) [inline pkg/kubectl/resource/visitor.go]: Please add a comment explaining what "Source" means. Is this the name of a .go file? a .json/.yaml file?

File: pkg/kubectl/cmd/create.go (hunk 1 above)
Comment (lavalamp) [inline pkg/kubectl/cmd/create.go]: It's not clear to me who calls this and whether or not it's important for these errors to be introspectable.

Comment (lavalamp) [inline pkg/kubectl/cmd/create.go]: Well-- this is a public method, so anyone could call it, no? CheckErr isn't guaranteed to be the thing receiving the error?

Comment (jlowdermilk) [inline pkg/kubectl/cmd/create.go]: Technically anyone could call it, but in practice it's only ever called in NewCmdCreate and in tests. Come to think of it, all the Run* functions should actually be private.

**Comment by jlowdermilk** [comment]
cc @smarterclayton

File: pkg/kubectl/cmd/create.go (hunk 1 above)
Comment (lavalamp) [inline pkg/kubectl/cmd/create.go]: Well, it's kinda handy to be able to call kubectl functions from e.g. tests, so let's think carefully before we do that... :)

**Comment by smarterclayton** [comment]
LGTM too

File: pkg/kubectl/resource/visitor.go [hunk 3]
```diff
@@ -66,6 +66,9 @@ type Info struct {
 	Namespace string
 	Name      string
 
+	//Optional, Source is the filename or URL to template file (.json or .yaml),
+	//or stdin to use to handle the resource
```
Comment (lavalamp) [inline pkg/kubectl/resource/visitor.go]: Nit: A single space after //, please.

File: pkg/kubectl/cmd/util/helpers.go [hunk 4]
```diff
@@ -48,6 +48,16 @@ type debugError interface {
 	DebugError() (msg string, args []interface{})
 }
 
+// AddSourceToErr adds handleResourcePrefix and source string to error message.
+// handleResourcePrefix is the string like "creating", "deleting" etc.
+// souce is the filename or URL to the template file(*.json or *.yaml), or stdin to use to handle the resource.
+func AddSourceToErr(handleResourcePrefix string, source string, err error) error {
```
Comment (lavalamp) [inline pkg/kubectl/cmd/util/helpers.go]: s/handleResourcePrefix/verb/

File: pkg/kubectl/cmd/delete.go [hunk 5]
```diff
@@ -122,14 +122,14 @@ func ReapResult(r *resource.Result, f *cmdutil.Factory, out io.Writer, isDefault
 			if kubectl.IsNoSuchReaperError(err) && isDefaultDelete {
 				return deleteResource(info, out)
 			}
-			return err
+			return cmdutil.AddSourceToErr("Reaper", info.Source, err)
```
Comment (lavalamp) [inline pkg/kubectl/cmd/delete.go]: "reaping"

**Comment by feihujiang** [comment]
@lavalamp I updated according to your advice, Could you please review this PR again?

**Comment by lavalamp** [comment]
LGTM

**Comment by lavalamp** [comment]
@dchen1107 for ok-to-merge?

**Comment by dchen1107** [comment]
LGTM

**Pull request merged**